crates/solver-service/src/apis/order.rs

```diff
@@ -0,0 +1,294 @@
+	let settlement_type = SettlementType::Escrow;
+
+	// Create settlement data from the raw order data
+	let settlement_data = serde_json::json!({
+		"raw_order_data": order.data.get("raw_order_data").cloned().unwrap_or_else(|| serde_json::json!(null)),
+		"signature": order.data.get("signature").cloned().unwrap_or_else(|| serde_json::json!(null)),
+		"nonce": order.data.get("nonce").cloned().unwrap_or_else(|| serde_json::json!(null)),
+		"expires": order.data.get("expires").cloned().unwrap_or_else(|| serde_json::json!(null))
+	});
+
+	// Derive status from SolverEvent state stored in the system
+	let status = derive_order_status_from_events(&order, solver).await?;
+
+	// Try to retrieve fill transaction hash from storage
+	let fill_transaction = if let Ok(fill_tx_hash) = solver
+		.storage()
+		.retrieve::<solver_types::TransactionHash>("fills", &order.id)
+		.await
+	{
+		// Create a structured fill transaction object
+		Some(serde_json::json!({
+			"hash": format!("0x{}", alloy_primitives::hex::encode(&fill_tx_hash.0)),
+			"status": "confirmed", // or derive from actual status
+			"timestamp": chrono::Utc::now().timestamp() // you might want to store actual timestamp
+		}))
+	} else {
+		// Fall back to existing data in order.data if available
+		order.data.get("fillTransaction").cloned()
+	};
+
+	let response = OrderResponse {
+		id: order.id,
+		status,
+		created_at: order.created_at,
+		last_updated: chrono::Utc::now().timestamp() as u64,
+		quote_id: None, // EIP-7683 orders don't have quote IDs by default
+		input_amount,
+		output_amount,
+		settlement_type,
+		settlement_data,
+		fill_transaction,
+		error_details: order
+			.data
+			.get("errorDetails")
+			.and_then(|v| v.as_str())
+			.map(|s| s.to_string()),
+	};
+
+	Ok(response)
+}
+
+/// Derives OrderStatus from the current state of the order in the event-driven system.
+async fn derive_order_status_from_events(
+	order: &solver_types::Order,
+	solver: &SolverEngine,
+) -> Result<OrderStatus, GetOrderError> {
+	let order_id = &order.id;
+
+	// Check if order has been completed (claim transaction confirmed)
+	if let Ok(_completion_time) = solver
+		.storage()
+		.retrieve::<u64>("completed_orders", order_id)
+		.await
+	{
+		return Ok(OrderStatus::Finalized);
+	}
+
+	// Also check for completion status in order data (updated by core solver)
+	if let Some(status) = order.data.get("status") {
+		if status.as_str() == Some("Finalized") {
+			return Ok(OrderStatus::Finalized);
+		}
+	}
+
+	// Check if the order has a fill transaction (indicates execution started)
+	if let Ok(_fill_tx_hash) = solver
+		.storage()
+		.retrieve::<solver_types::TransactionHash>("fills", order_id)
+		.await
+	{
+		// Order has been executed (fill transaction submitted) but not yet finalized
+		return Ok(OrderStatus::Executed);
+	}
+
+	// Check if order has pending execution
+	if let Ok(_) = solver
+		.storage()
+		.retrieve::<(solver_types::Order, solver_types::ExecutionParams)>(
+			"pending_execution",
```

Comment:
I feel like we could have 1 orders table and track the status from there instead of 3-4 different tables..?